test-data/unit/check-typevar-tuple.test

```diff
@@ -2628,3 +2628,11 @@ def fn(f: Callable[[*tuple[T]], int]) -> Callable[[*tuple[T]], int]: ...
 def test(*args: Unpack[tuple[T]]) -> int: ...
 reveal_type(fn(test))  # N: Revealed type is "def [T] (T`1) -> builtins.int"
 [builtins fixtures/tuple.pyi]
+
+[case testNoCrashSubclassingTUpleWithTrivialUnpack]
+# https://github.com/python/mypy/issues/19105
+from typing import Unpack
+
+class A(tuple[Unpack[tuple[int]]]): ...
```

Comment:
Could you please also add a test for
```python
class C(tuple[Unpack[tuple[int, ...]]]): ...
```
(unless there is already such test case)? I am not sure we properly handle such types. Also maybe test not just class creation, but some tuple ops as well, such as e.g. unpacking etc.

Reply:
Surprisingly we don't have such a test too, but it was already working before my patch - `*tuple[int, ...]` is not a trivial unpack. But your feeling is overall correct - I'm able to make it crash with 

```python
a: A
tuple(a)
```

and also get a terrible diagnostic on

```python
b: B
_, = b  # E: Variadic tuple unpacking requires a star target
```

despite B clearly not being variadic. I'll move this to draft for now, there's little benefit from merging a crash fix that causes a next one immediately...